crates/cache-key/src/canonical_url.rs

```diff
@@ -32,8 +32,8 @@ impl CanonicalUrl {
         }
 
         // Strip credentials.
-        url.set_password(None).unwrap();
-        url.set_username("").unwrap();
+        let _ = url.set_password(None);
+        let _ = url.set_username("");
```

Comment:
Wouldn't we return on line 31 if there's no host? or line 26 if there cannot be a base? Weird.